src/Aggregate/AggregateTranslatorConfiguration.php

```diff
@@ -0,0 +1,286 @@
+<?php
+/**
+ * This file is part of the prooph/event-store.
+ * (c) 2014-2016 prooph software GmbH <contact@prooph.de>
+ * (c) 2015-2016 Sascha-Oliver Prolic <saschaprolic@googlemail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Prooph\EventStore\Aggregate;
+
+use Assert\Assertion;
+
+class AggregateTranslatorConfiguration
+{
+    const DEFAULT_VERSION_METHOD_NAME = 'getVersion';
+    const DEFAULT_IDENTIFIER_METHOD_NAME = 'getId';
+    const DEFAULT_POP_RECORDED_EVENTS_METHOD_NAME = 'popRecordedEvents';
+    const DEFAULT_REPLAY_EVENTS_METHOD_NAME = 'replay';
+    const DEFAULT_STATIC_RECONSTITUTE_FROM_HISTORY_METHOD_NAME = 'reconstituteFromHistory';
+    const DEFAULT_EVENT_TO_MESSAGE_CALLBACK = null;
+    const DEFAULT_MESSAGE_TO_EVENT_CALLBACK = null;
+
+    /**
+     * @var string
+     */
+    private $versionMethodName;
+
+    /**
+     * @var string
+     */
+    private $identifierMethodName;
+
+    /**
+     * @var string
+     */
+    private $popRecordedEventsMethodName;
+
+    /**
+     * @var string
+     */
+    private $replayEventsMethodName;
+
+    /**
+     * @var string
+     */
+    private $staticReconstituteFromHistoryMethodName;
+
+    /**
+     * @var null|callable
+     */
+    private $eventToMessageCallback;
+
+    /**
+     * @var null|callable
+     */
+    private $messageToEventCallback;
+
+    /**
+     * AggregateTranslatorConfiguration constructor.
+     *
+     * @param $versionMethodName
+     * @param $identifierMethodName
+     * @param $popRecordedEventsMethodName
+     * @param $replayEventsMethodName
+     * @param $staticReconstituteFromHistoryMethodName
+     * @param null $eventToMessageCallback
+     * @param null $messageToEventCallback
+     */
+    public function __construct(
+        $versionMethodName,
+        $identifierMethodName,
+        $popRecordedEventsMethodName,
+        $replayEventsMethodName,
+        $staticReconstituteFromHistoryMethodName,
+        $eventToMessageCallback = null,
+        $messageToEventCallback = null
+    ) {
+        $this->versionMethodName = $versionMethodName;
```

Comment:
Type assertions aren't missing, they are made within the dedicated with* methods for each property. I think this shouldn't be a problem since the constructor of configuration is private and only called within the factory method for default configuration.